cves/CVE-2013-0868.yml

```diff
@@ -1,368 +1,406 @@
+    a quick explanation of how it applies.
+
+    Don't feel the need to claim that ALL of these apply, but it's pretty likely
+    that one or two of them apply.
+
+    If you think of another lesson we covered in class that applies here, feel
+    free to give it a small name and add one in the same format as these.
+  defense_in_depth:
+    applies: true
+    note: Return codes from functions that can error should be checked to provide failover layers, but weren't in this case.
+  least_privilege:
+    applies:
+    note:
+  frameworks_are_optional:
+    applies:
+    note:
+  native_wrappers:
+    applies:
+    note:
+  distrust_input: 
+    applies: true
+    note: The parser trusts the input and assumes it is properly form and has a length over 0.
+  security_by_obscurity:
+    applies:
+    note:
+  serial_killer:
+    applies:
+    note:
+  environment_variables:
+    applies:
+    note:
+  secure_by_default:
+    applies:
+    note:
+  yagni:
+    applies:
+    note:
+  complex_inputs: 
+    applies: true
+    note: Video files are complex data structures and each codec introduces unique fields and data formats.
+mistakes:
+  question: |
+    In your opinion, after all of this research, what mistakes were made that
+    led to this vulnerability? Design mistakes? Maintainability? Requirements?
+    Miscommunications? Lack of testing? Lack of understanding? Lack of
+    specifications? Working alone?
+
+    An answer like "it was just a coding mistake" is not thoughtful enough.
+    If it's such an easy mistake to make, how was it missed?
+
+    Also, look at the CWE entry for this vulnerability and examine the mitigations
+    they have written there. Are they doing those? Does the fix look proper?
+
+    Use those questions to inspire your answer. Don't feel obligated to answer
+    every one. Write a thoughtful entry here that those ing the software
+    engineering industry would find interesting.
+  answer: |
+    This vulnerability occured due to a mix of a lack of testing and a lack of review.
+    Some measures already existed in this video decoder to indicate a parsing failure,
+    however they were not checked. Other checks were lacking completely, such as looking
+    if the file length was greater than 0. These are issues that should have been caught
+    with a manual review or testing, as they are quite basic error handling/input validation
+    issues.
+    Further, added codecs lack propery ongoing testing. Countless refactors/splits occured
+    in this file and other codecs in the many years between the vcc and fix, but no one seemed
+    to question the functionality or retest it. ffmpeg codecs seem to have few maintainers who
+    understand each, relying too much on certain knowledgable individuals to find and fix such
+    issues eventually. The proper mitigations for an out of bounds write were made, but this
+    vulnerability existed for more than half a decade, indicating issues with code testing.

```

Comment:
Excellent. This is really good. I'm going to tag it. I'm not able to review the next file but I'm going to tag this as great work.